pkg/controller/pxc/users.go

```diff
@@ -0,0 +1,136 @@
+package pxc
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"os"
+
+	api "github.com/percona/percona-xtradb-cluster-operator/pkg/apis/pxc/v1"
+	"github.com/percona/percona-xtradb-cluster-operator/pkg/pxc/users"
+	"github.com/pkg/errors"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (r *ReconcilePerconaXtraDBCluster) reconcileUsers(cr *api.PerconaXtraDBCluster) error {
+	if cr.Status.Status != api.AppStateReady {
+		return nil
+	}
+
+	if len(cr.Spec.Users.Secrets) == 0 {
+		return nil
+	}
+
+	for _, secretName := range cr.Spec.Users.Secrets {
+		err := r.handleUsersSecret(secretName, cr)
+		if err != nil {
+			log.Error(err, "handle users secret "+secretName)
+		}
+	}
+
+	return nil
+}
+
+func (r *ReconcilePerconaXtraDBCluster) handleUsersSecret(secretName string, cr *api.PerconaXtraDBCluster) error {
+	usersSecretObj := corev1.Secret{}
+	err := r.client.Get(context.TODO(),
+		types.NamespacedName{
+			Namespace: cr.Namespace,
+			Name:      secretName,
+		},
+		&usersSecretObj,
+	)
+	if err != nil && k8serrors.IsNotFound(err) {
+		return nil
+	} else if err != nil {
+		return errors.Wrapf(err, "get users secret '%s'", secretName)
+	}
+
+	newHash := ""
+	if secretData, ok := usersSecretObj.Data["secret.yaml"]; ok {
+		newHash = sha256Hash(secretData)
+	}
+	lastAppliedHash := ""
+	if hash, ok := usersSecretObj.Annotations["last-applied"]; ok {
+		lastAppliedHash = hash
+		if lastAppliedHash == newHash {
+			if usersSecretObj.Annotations["status"] == "succeded" || usersSecretObj.Annotations["status"] == "failed" {
+				return nil
+			}
+		}
+	}
+
+	if len(usersSecretObj.Annotations) == 0 {
+		usersSecretObj.Annotations = make(map[string]string)
+	}
+	usersSecretObj.Annotations["last-applied"] = newHash
+	usersSecretObj.Annotations["status"] = "applying"
+	err = r.client.Update(context.TODO(), &usersSecretObj)
+	if err != nil {
+		return errors.Wrap(err, "update secret last-applied")
+	}
+
+	operatorPod := corev1.Pod{}
+	err = r.client.Get(context.TODO(),
+		types.NamespacedName{
+			Namespace: cr.Namespace,
+			Name:      os.Getenv("HOSTNAME"),
+		},
+		&operatorPod,
+	)
+	if err != nil {
+		return errors.Wrap(err, "get operator deployment")
+	}
+	containerImage := operatorPod.Spec.Containers[0].Image
+	imagePullSecrets := operatorPod.Spec.ImagePullSecrets
+	job := users.Job(cr, secretName, newHash)
+	job.Spec = users.JobSpec(secretName, containerImage, job, cr, imagePullSecrets)
```

Comment:
I believe this code is needed only if we Create new job (between lines 94-95).
Job name can be calculated outside of `Job` function

Reply:
done https://github.com/percona/percona-xtradb-cluster-operator/pull/342/commits/1614b84316670d6c6c63719f51d8ccbea626a485

Reply:
done https://github.com/percona/percona-xtradb-cluster-operator/pull/342/commits/1614b84316670d6c6c63719f51d8ccbea626a485